scripts/dev/utils/git_commands.py

```
@@ -0,0 +1,204 @@
+
+
+"""
+Utility functions for git operations.
```

Why not use an existing python git library (e.g., GitPython) instead of rolling out our own?

---

GitPython doesn’t cover all the cases we need. For example, commands like gh or git send-email aren’t supported, and even for common commands, not all flags are exposed (e.g., some options for git commit --amend or git merge --no-ff/--signoff  are missing). GitPython also raises exceptions when a command fails instead of returning a status code, which makes it harder to uniformly handle errors in scripts. By using git_commands wrapper, we always get a (return_code, output) tuple, giving us full flexibility and consistency across both standard Git commands and less common ones.